### PR TITLE
Adding runtime guard to protect drop overload  feature

### PR DESCRIPTION
**Pull request by yanjunxiang-google**
The runtime guard is default to be false.

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/common/router/router.cc [hunk 1]
```diff
@@ -2026,15 +2026,18 @@ uint32_t Filter::numRequestsAwaitingHeaders() {
 
 bool Filter::checkDropOverload(Upstream::ThreadLocalCluster& cluster,
                                std::function<void(Http::ResponseHeaderMap&)>& modify_headers) {
+  if (!Runtime::runtimeFeatureEnabled("envoy.reloadable_features.enable_drop_overload")) {
```
Comment (adisuissa) [inline source/common/router/router.cc:2029]: Please add this as part of the `if (cluster.dropOverload().value())` below.
This will couple the change with the code (so if later new code is added after the `if` scope, it will be used whether the runtime feature is enabled or not).

File: source/common/router/router.cc [hunk 2]
```diff
@@ -2026,15 +2026,18 @@ uint32_t Filter::numRequestsAwaitingHeaders() {
...
+    return false;
+  }
   if (cluster.dropOverload().value()) {
     ENVOY_STREAM_LOG(debug, "Router filter: cluster DROP_OVERLOAD configuration: {}", *callbacks_,
                      cluster.dropOverload().value());
     if (config_.random_.bernoulli(cluster.dropOverload())) {
       ENVOY_STREAM_LOG(debug, "The request is dropped by DROP_OVERLOAD", *callbacks_);
       callbacks_->streamInfo().setResponseFlag(StreamInfo::ResponseFlag::DropOverLoad);
-      chargeUpstreamCode(Http::Code::ServiceUnavailable, nullptr, true);
+      chargeUpstreamCode(Http::Code::BadGateway, nullptr, true);
```
Comment (adisuissa) [inline source/common/router/router.cc:2038]: Seems not related to this change.

Comment (yanjunxiang-google) [reply source/common/router/router.cc:2038]: I closed this PR.  The new PR to protect drop_overload is https://github.com/envoyproxy/envoy/pull/31809/.  However, please hold off reviewing that PR until we decided what's the right approach.

File: source/common/router/router.cc (hunk 1 above)
Comment (yanjunxiang-google) [reply source/common/router/router.cc:2029]: I separate it just for a little bit easy to read.

I don't fully understand your comments on why it is better to combine the two if-conditions together.
